### ISSUE TITLE
Aggressive Float/Double Optimizations, e.g. `! ( x <= 0 )`  ⟶ `x > 0`

### ISSUE THREAD

**Issue by DirkToewe** [scala-js/scala-js#3575]
The follwing code prints `true` on the JVM and `false` on Node:
```scala
def f( x: Double ) = ! (x <= 0)

// ...

println( f(Double.NaN) )
```
It appears that ScalaJS aggressively optimizes `! (x <= 0)` to `x > 0`. With `scalaJSOptimizerOptions ~= (_ withDisableOptimizer true)`, the Node result is consistent with the JVM.

**Comment by sjrd** [comment]
Ugh! I can't believe I let that slip in our own optimizer, when I had already discovered a similar problem in scalac itself :man_facepalming: 

It's there: https://github.com/scala-js/scala-js/blob/aad1591556a5b49d22d32a17650daa97d14aae68/linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala#L2695-L2700
Those cases should be removed. We can never optimize `!(l ? r)` when `?` is a Double comparison.

**Comment by DirkToewe** [comment]
I believe You can optimize `! ( x == y )` and `! ( x != y )` but I'm not a 100% sure.

**Comment by sjrd** [comment]
Yes, indeed.

**Comment by sjrd** [comment]
Fixed in 5de468a09cba7c8711a25dbebff8a31d0793cbbe.
